From b79f79448a70b5718db2428939e00150315b5eda Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Wed, 6 Aug 2008 20:46:42 +0100
Subject: [PATCH] add-buildkeep-hash-named-copy.patch

Keep a copy of last "moredrivers" config kernel built stamped with
branch and git head at build time.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 build |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/build b/build
index 9c2d561..63bb0fc 100755
--- a/build
+++ b/build
@@ -15,7 +15,15 @@ fi
 
 if make -j5 ARCH=arm EXTRAVERSION=$VERSION; then
 	${CROSS_COMPILE}objcopy -O binary -R .note -R .comment -S arch/arm/boot/compressed/vmlinux linux.bin
-	mkimage -A arm -O linux -T kernel -C none -a 30008000 -e 30008000 -n "Openmoko Kernel Image Neo1973(GTA02)" -d linux.bin uImage.bin
+	mkimage -A arm -O linux -T kernel -C none -a 30008000 -e 30008000 -n "Openmoko Freerunner Kernel" -d linux.bin uImage.bin
+
+	# we can see if it is an "moredrivers" build by looking for USB Eth gadget
+	# if it is then keep a stamped copy of last build
+
+	if [ ! -z "`grep CONFIG_USB_ETH=y .config`" ] ; then
+		rm -f uImage-moredrivers*
+		cp uImage.bin uImage-moredrivers$VERSION.bin
+	fi
 	exit 0
 else
 	exit 1
-- 
1.5.6.5

